lib/logstash/inputs/file.rb

```diff
@@ -199,15 +199,15 @@ def run(queue)
         queue << event
       end
     end
-    finished
   end # def run
 
   public
-  def teardown
-    if @tail
-      @tail.sincedb_write
-      @tail.quit
-      @tail = nil
-    end
-  end # def teardown
+  def stop
+    super
+    @tail.quit if @tail # _sincedb_write is called implicitly
```

Comment:
@purbon the answer is here https://github.com/jordansissel/ruby-filewatch/blob/master/lib/filewatch/watch.rb#L115 .

So, it appears that it does not immediately stop the subscriptions by force, but rather sets a var that will be noticed next time around the loop. I think we're good here.